rllib/policy/policy.py

```diff
@@ -855,13 +892,28 @@ def on_global_var_update(self, global_vars: Dict[str, TensorType]) -> None:
             self.global_timestep = global_vars["timestep"]
 
     @DeveloperAPI
-    def export_checkpoint(self, export_dir: str) -> None:
-        """Export Policy checkpoint to local directory.
+    def export_checkpoint(
```

Comment:
Generic `export_checkpoint` method (works for all DL frameworks and for all new PolicyV2 classes).

Creates checkpoint on disk AND return AIR Checkpoint object.

Reply:
A question, why doesn't Algorithm call this function to save individual policies in all these sub-dirs?
something like
```
class Algorithm:
    ...
    def save_checkpoint(dir):
        ... saves Algorithm state in dir ...
        for id, policy in local_worker.policy_map.items():
            policy.export_checkpoint(path.join(dir, id))
```
instead, Algorithm would dump all those states for the policies ...

Reply:
Yeah, valid point. The thing is that `Algorithm` uses its own `__getstate__` method, which already returns all policies' states in the dict. Now calling `Policy.export_checkpoint` on top of that would force us to compile each policy's state twice (once in `Algorithm.__getstate__` and once in `Policy.export_checkpoint()`). I feel like this would be overkill.
This is why we need cleaner naming conventions, imo:
* `save()`/`export_checkpoint()` -> write stuff to disk
* `get_state()` -> return a dict; nothing written to disk

We could argue now that `Policy.export_checkpoint` should be renamed into `Policy.save()`
🤷‍♂️

Reply:
b/c `save()` is the method that writes stuff to disk.

Reply:
I actually fixed this now, as you suggested. But I had to add a new arg to tell the Policy's `export_checkpoint` method to NOT regenerate the state, if this state is already provided.

Reply:
ok, sounds good. thanks.